crates/assists/src/handlers/infer_function_return_type.rs

```diff
@@ -0,0 +1,237 @@
+use hir::HirDisplay;
+use syntax::{ast, AstNode, SyntaxToken, TextSize};
+use test_utils::mark;
+
+use crate::{AssistContext, AssistId, AssistKind, Assists};
+
+// Assist: infer_function_return_type
+//
+// Adds the return type to a function or closure inferred from its tail expression if it doesn't have a return
+// type specified.
+//
+// ```
+// fn foo() { 4<|>2i32 }
+// ```
+// ->
+// ```
+// fn foo() -> i32 { 42i32 }
+// ```
+pub(crate) fn infer_function_return_type(acc: &mut Assists, ctx: &AssistContext) -> Option<()> {
```

Comment:
Might also be cool to support the assist in the following cases:

```rust
fn foo() <|> {}
fn foo() -> <|>_ {}
```

Reply:
That's a nice idea